src/mlpack/core/optimizers/fw/atoms.hpp

```diff
@@ -0,0 +1,249 @@
+/**
+ * @file atoms.hpp
+ * @author Chenzhe Diao
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_FW_ATOMS_HPP
+#define MLPACK_CORE_OPTIMIZERS_FW_ATOMS_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "func_sq.hpp"
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Class to hold the information and operations of current atoms in the
+ * soluton space.
+ */
+class Atoms
+{
+ public:
+  Atoms(){ /* Nothing to do. */ }
+
+  /**
+   * Add atom into the solution space.
+   *
+   * @param v new atom to be added.
+   * @param c coefficient of the new atom.
+   */
+  void AddAtom(const arma::mat& v, const double c = 0)
+  {
+    if (currentAtoms.is_empty())
+    {
+      CurrentAtoms() = v;
+      CurrentCoeffs() = c;
+    }
+    else
+    {
+      currentAtoms.insert_cols(0, v);
+      arma::vec cVec = {c};
```

Comment:
Since scrolling through the appveyor log is kinda tedious, I usually download the log and search for "error ", note the extra space. Here is the interesting part:

```
[00:16:31] c:\projects\mlpack\src\mlpack\core\optimizers\fw\atoms.hpp(45): error C2440: 'initializing': cannot convert from 'initializer list' to 'arma::Col<double>' [C:\projects\mlpack\build\src\mlpack\tests\mlpack_test.vcxproj]
[00:16:31]   c:\projects\mlpack\src\mlpack\core\optimizers\fw\atoms.hpp(45): note: Constructor for class 'arma::Col<double>' is declared 'explicit'
[00:16:31] c:\projects\mlpack\src\mlpack\core\optimizers\fw\constr_lpball.hpp(105): warning C4244: 'argument': conversion from 'double' to 'const arma::uword', possible loss of data [C:\projects\mlpack\build\src\mlpack\tests\mlpack_test.vcxproj]
[00:16:32] C:\projects\mlpack\src\mlpack\tests\frankwolfe_test.cpp(42): error C2440: 'initializing': cannot convert from 'initializer list' to 'arma::Col<double>' [C:\projects\mlpack\build\src\mlpack\tests\mlpack_test.vcxproj]
[00:16:32]   C:\projects\mlpack\src\mlpack\tests\frankwolfe_test.cpp(42): note: No constructor could take the source type, or constructor overload resolution was ambiguous
[00:16:32] C:\projects\mlpack\src\mlpack\tests\frankwolfe_test.cpp(98): error C2440: 'initializing': cannot convert from 'initializer list' to 'arma::Mat<double>' [C:\projects\mlpack\build\src\mlpack\tests\mlpack_test.vcxproj]
[00:16:32]   C:\projects\mlpack\src\mlpack\tests\frankwolfe_test.cpp(98): note: No constructor could take the source type, or constructor overload resolution was ambiguous
[00:16:32] C:\projects\mlpack\src\mlpack\tests\frankwolfe_test.cpp(101): error C2440: 'initializing': cannot convert from 'initializer list' to 'arma::Col<double>' [C:\projects\mlpack\build\src\mlpack\tests\mlpack_test.vcxproj]
[00:16:32]   C:\projects\mlpack\src\mlpack\tests\frankwolfe_test.cpp(101): note: No constructor could take the source type, or constructor overload resolution was ambiguous
[00:16:32] C:\projects\mlpack\src\mlpack\tests\frankwolfe_test.cpp(124): error C2440: 'initializing': cannot convert from 'initializer list' to 'arma::Col<double>' [C:\projects\mlpack\build\src\mlpack\tests\mlpack_test.vcxproj]
[00:16:32]   C:\projects\mlpack\src\mlpack\tests\frankwolfe_test.cpp(124): note: No constructor could take the source type, or constructor overload resolution was ambiguous
```

Looks like using ``CurrentCoeffs().fill(c)`` instead of ``CurrentCoeffs() = c;`` would solve the problem the same applies to ``arma::vec cVec = {c};``.